src/components/Header/Header.tsx

```diff
@@ -34,6 +34,31 @@ export default function Header({
 }) {
   const { logout, session } = useContext(SessionContext);
 
+  const isUrl = (candidateUrl: string | URL) => {
+    try {
+      // If url is not URL-shaped, this will throw.
+      // eslint-disable-next-line no-new
+      new URL(candidateUrl);
+      return true;
+    } catch (_e) {
+      return false;
+    }
+  };
+
+  const handleLogout = async () => {
+    if (
+      session.info.clientAppId !== undefined &&
+      isUrl(session.info.clientAppId)
+    ) {
+      await logout({
+        logoutType: "idp",
+        postLogoutUrl: new URL("/", window.location.href).toString(),
```

Comment:
```suggestion
        postLogoutUrl: new URL(window.origin).href,
```
This must be aligned with the way we construct the post-logout URL in `buildClientIdentifierDoc`. It's a problem that it is based on `window`, it should be driven by the environment, but let's not tackle this here, because it has a broader scope than this PR.